pkg/connector/aws_account.go

```diff
@@ -557,3 +572,306 @@ func (o *accountResourceType) getOktaAppGroupFromCacheOrFetch(ctx context.Contex
 
 	return appGroupSAMLRoles, nil
 }
+
+const apiPathApplicationGroup = "/api/v1/apps/%s/groups/%s"
+
+type JsonPatchOperation struct {
+	// The operation (PATCH action)
+	Op string `json:"op,omitempty"`
+	// The resource path of the attribute to update
+	Path string `json:"path,omitempty"`
+	// The update operation value
+	Value interface{} `json:"value,omitempty"`
+}
+
+func (o *accountResourceType) Grant(ctx context.Context, principal *v2.Resource, entitlement *v2.Entitlement) (annotations.Annotations, error) {
+	if principal.Id.ResourceType != resourceTypeUser.Id && principal.Id.ResourceType != resourceTypeGroup.Id {
+		return nil, fmt.Errorf("okta-aws-connector: only users or groups can be granted app membership")
+	}
+	awsConfig, err := o.connector.getAWSApplicationConfig(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("error getting aws app settings config")
+	}
+
+	if awsConfig.UseGroupMapping {
+		return nil, fmt.Errorf("okta-aws-connector: group assignments are based on group names matching regex")
+	}
+
+	appID := o.connector.awsConfig.OktaAppId
+	newSamlRole := entitlement.GetSlug()
```

Comment:
this seems weird to me?  do we always have a full entitlement object when provisioning like this? I'd kinda assume less so, and Slug() is something you can rename in the UI right?